File: bridge/typescript/src/_impl/dir/source/comment.ts
import type { CommentNewlines } from "../../../_generated/dir/source/comment.js";

/** Bit flag for a leading newline. */
const LEADING = 1 << 0;
/** Bit flag for a trailing newline. */
const TRAILING = 1 << 1;

export const CommentNewlinesImpl = {
    /** Return whether a comment has a leading newline. */
    hasLeadingNewline(newlines: CommentNewlines): boolean {
        return (newlines.bits & LEADING) !== 0;
    },

    /** Return whether a comment has a trailing newline. */
    hasTrailingNewline(newlines: CommentNewlines): boolean {
        return (newlines.bits & TRAILING) !== 0;
    },
};

File: bridge/python/src/destack/_impl/dir/source/comment.py
from __future__ import annotations

from destack._impl.model import ModelImpl

LEADING = 1 << 0
TRAILING = 1 << 1


class CommentNewlinesImpl(ModelImpl):
    """Methods for comment newline flags."""

    def has_leading_newline(self) -> bool:
        """Return whether a comment has a leading newline."""

        return self.bits & LEADING != 0

    def has_trailing_newline(self) -> bool:
        """Return whether a comment has a trailing newline."""

        return self.bits & TRAILING != 0
